### PR TITLE
rgw: change common hex_to_buf tool-function type check

### PR DESCRIPTION
**Pull request by imtzw**
protect type checker from complaining

Signed-off-by: tongzhiwei <tongzhiwei_yewu.cmss.chinamobile.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: src/rgw/rgw_common.h [hunk 1]
```diff
@@ -1384,7 +1384,7 @@ static inline int hexdigit(char c)
   return -EINVAL;
 }
 
-static inline int hex_to_buf(const char *hex, char *buf, int len)
+static inline int hex_to_buf(const char *hex, unsigned char *buf, int len)
```
Comment (mattbenjamin) [inline src/rgw/rgw_common.h:1387]: Seems valid, though I think treating char as signed is harmless in this sort of (C) code

Comment (imtzw) [reply src/rgw/rgw_common.h:1387]: yes it is infact harmless . just matters compile time

**Comment by cbodley** [comment]
from the 'make check' log at https://jenkins.ceph.com/job/ceph-pull-requests/111658/consoleFull:
```
src/rgw/rgw_op.cc:6535:5: error: no matching function for call to 'hex_to_buf'
    hex_to_buf(partetag.c_str(), petag, CEPH_CRYPTO_MD5_DIGESTSIZE);
    ^~~~~~~~~~
/home/jenkins-build/build/workspace/ceph-pull-requests/src/rgw/rgw_common.h:1387:19: note: candidate function not viable: no known conversion from 'char[16]' to 'unsigned char *' for 2nd argument
static inline int hex_to_buf(const char *hex, unsigned char *buf, int len)
                  ^
```

**Comment by imtzw** [comment]
seems many usages of this tool function have to be adjusted

**Pull request closed without merging**